src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -123,23 +217,59 @@ private void buildInnerHit(List<String> fieldNames, NestedQueryBuilder query) {
      * Why linear search? Because NestedQueryBuilder hides "path" field from any access.
      * Assumption: collected NestedQueryBuilder list should be very small or mostly only one.
      */
-    private NestedQueryBuilder findNestedQueryWithSamePath(List<NestedQueryBuilder> nestedQueries, String path) {
+    private NestedQueryBuilder findNestedQueryWithSamePath(
+            List<NestedQueryBuilder> nestedQueries, String path) {
+
         return nestedQueries.stream().
-                             filter(query -> isSamePath(path, query)).
-                             findAny().
-                             orElseGet(createEmptyNestedQuery(path));
+                filter(query -> isSamePath(path, query)).
+                peek(x -> setLastPath(path)).
+                findAny().
+                orElseGet(createNestedQuery(path));
     }
 
     private boolean isSamePath(String path, NestedQueryBuilder query) {
```

Comment:
could you add a comment on this method explaining why paths are equal only if queries are equal?

Reply:
Sure I will, but this method is not my change. Some of my understanding:  there is no way to get the path from `query`, which is an instance of `NestedQueryBuider`. It just doesn't have a `getter` method for path, and the `path` variable can not be accessed. But any NestedQueryBuilder is initialized by a path, so here it compares the path by first creating a tmp query and then compare with the existing query.